packages/mds-ingest-service/repository/index.ts

```diff
@@ -355,6 +356,48 @@ class IngestReadWriteRepository extends ReadWriteRepository {
   public getEventsUsingCursor = async (cursor: string): Promise<GetVehicleEventsResponse> =>
     this.getEvents(this.parseCursor(cursor))
 
+  public getTripEvents = async (params: ReadTripEventsQueryParams) => {
+    const { skip, take = 100, start_time, end_time, provider_id } = params
+    try {
+      const connection = await this.connect('rw')
+      const tripIdQuery = await connection
+        .getRepository(EventEntity)
+        .createQueryBuilder()
+        .distinctOn(['trip_id'])
+        .limit(take)
+      if (start_time) tripIdQuery.where('start_time >= :start_time', { start_time })
+      if (end_time) tripIdQuery.andWhere('end_time <= :end_time', { end_time })
+      if (provider_id) tripIdQuery.andWhere('provider_id = :provider_id', { provider_id })
+      if (skip) tripIdQuery.andWhere('trip_id > :trip_id', { skip })
```

Comment:
If `start_time` is undefined, will these `andWhere` clauses work properly?

Reply:
they do! thankfully. I tried a few more configurations, and this is the least awkward.